store/data_source.go

```diff
@@ -43,6 +45,28 @@ func (s *DataSourceService) CreateDataSource(ctx context.Context, create *api.Da
 		return nil, FormatError(err)
 	}
 
+	// Upsert instance.dataSourceList cache after created its related datasource.
+	if dataSource != nil {
+		instance := &api.Instance{}
+		has, err := s.cache.FindCache(api.InstanceCache, dataSource.InstanceID, instance)
```

Comment:
As a 2nd thought, this might lead to tricky concurrency bug.

Say another thread is also fetching the instance cache, then one of the thread would overwrite the other thread's instance cache depending on the timing.

So I am now thinking to not store cache for data source at all. And just fetch the data source on demand every time we retrieve the instance.

Reply:
I have moved the fetch data source list of instance from store to server.